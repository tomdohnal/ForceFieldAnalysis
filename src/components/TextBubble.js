import { createComponent } from 'react-fela';

const arrowMargin = 12;
const arrowHeight = 12;

const textBubbleStyles = ({ color, left }) => ({
  position: 'relative',
  display: 'inline-block',
  padding: '4px 12px',
  margin: '0 0 16px',
  border: `2px solid ${color}`,
  color: '#333',
  background: '#fff',
  minWidth: '100px',
  WebkitBorderRadius: '5px',
  MozBorderRadius: '5px',
  borderRadius: '5px',
  ':before': {
    content: '""',
    position: 'absolute',
    bottom: `-${arrowHeight}px`,
    left: left ? `${arrowMargin}px` : `calc(100% - ${arrowMargin + (arrowHeight * 2)}px)`,
    borderWidth: `${arrowHeight}px ${arrowHeight}px 0`,
    borderStyle: 'solid',
    borderColor: `${color} transparent`,
    display: 'block',
    width: '0',
  },
  ':after': {
    content: '""',
    position: 'absolute',
    bottom: `-${arrowHeight - 3}px`,
    left: left ? `${arrowMargin + 3}px` : `calc(100% - ${arrowMargin + ((arrowHeight * 2) - 3)}px)`,
    borderWidth: `${arrowHeight - 3}px ${arrowHeight - 3}px 0`,
    borderStyle: 'solid',
    borderColor: '#fff transparent',
    display: 'block',
    width: '0',
  },
});

export default createComponent(textBubbleStyles, 'span');
