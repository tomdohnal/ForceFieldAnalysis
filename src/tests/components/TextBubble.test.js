import React from 'react';
import { shallow } from 'enzyme';

import TextBubble from '../../components/TextBubble';

describe('TextBubble', () => {
  it('renders correctly left TextBubble', () => {
    const tree = shallow(<TextBubble color="#FFF" left />);

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly right TextBubble', () => {
    const tree = shallow(<TextBubble color="#FFF" />);

    expect(tree).toMatchSnapshot();
  });
});