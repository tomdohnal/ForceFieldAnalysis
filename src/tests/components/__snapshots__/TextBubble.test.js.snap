// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBubble renders correctly left TextBubble 1`] = `
<textBubbleStyles
  color="#FFF"
  left={true}
  theme={Object {}}
/>
`;

exports[`TextBubble renders correctly right TextBubble 1`] = `
<textBubbleStyles
  color="#FFF"
  theme={Object {}}
/>
`;
